find_studies.py

```diff
@@ -0,0 +1,457 @@
+    print('exit\t\t(Quit annotating and save progress)')
+    print('-' * 80)
+
+
+def merge_study_groups(study):
+    '''Combine all study groups into a single group
+    
+    metaSRA splits samples from a study into multiple groups based on their metadata.
+    This function merges all the samples from a study back into one group
+    
+    Arguments
+    ---------
+    study: dict
+        The json-derived object containing information about a study
+
+    Returns
+    -------
+    all_groups: list
+        A list of dictionaries containing all experiments for a given study
+
+    '''
+    study_groups = study['sampleGroups']
+
+    all_groups = []
+    for group in study_groups:
+        for experiment in group['samples']:
+            all_groups.append(experiment)
+    return all_groups
+
+
+def print_rule_prompt():
+    ''' Print the prompt giving the user their options for creating a rule'''
+    print('-' * 80)
+    print('Rule options:')
+    print('case:\t\tAll samples containing this string are cases')
+    print('control:\tAll samples containing this string are controls')
+    print('invalid:\tAll samples containing this string are invalid (not RNA, not human, etc)')
+    print('study_info:\tPrint information about the study so it can be looked over in SRA')
+    print('done:\t\tStop writing rules for the study. All samples not assigned to case or control '
+          'will be saved as invalid')
+    print('restart:\tClear all previously written rules for this study and start over')
+    print('-' * 80)
+
+
+def get_study_info_as_xml(groups):
+    ''' Use a list of sample ids to get information about each sample from the SRA
+
+    Arguments
+    ---------
+    groups - dict
+        The MetaSRA info about all the samples for a given study
+
+    Returns
+    -------
+    root: xml.etree.ElementTree
+        An XML object containing the sample information returned by Entrez
+
+    '''
+    ids = []
+    for sample in groups:
+        # Get the SRX (experiment) id
+        ids.append(sample['experiments'][0]['id'])
+
+    ids = ','.join(ids)
+
+    # Get information about all the samples in the experiment
+    print('Querying Entrez...')
+
+    attempts = 0
+    while attempts < 10:
+        try:
+            result = requests.post('https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi', data={'db': 'sra', 'id': ids})
+
+            # Throw an exception if we get an HTTP error
+            result.raise_for_status()
+
+            # Parse the XML returned from Entrez
+            print('Parsing XML...')
+            out = open('test.txt', 'w')
+            out.write(strip_unicode(result.text))
+            root = ET.fromstring(strip_unicode(result.text))
+
+            return root
+        except requests.exceptions.HTTPError:
+            attempts += 1
+            print('Downloading dataset failed {} times'.format(attempts))
+            sleep(1)
```

Comment:
would suggest some sort of exponential backoff where the sleep increases as attempts grows.

Reply:
Increasing the amount of time sleeping is a good idea, I added a linear backoff though